core/templates/dev/head/services/UserService.js

```diff
@@ -17,16 +17,31 @@
  */
 
 oppia.factory('UserService', [
-  '$http', '$q', '$window', 'UrlInterpolationService', 'UserInfoObjectFactory',
-  'DEFAULT_PROFILE_IMAGE_PATH',
-  function($http, $q, $window, UrlInterpolationService, UserInfoObjectFactory,
-      DEFAULT_PROFILE_IMAGE_PATH) {
+  '$cookies', '$http', '$q', '$window', 'UrlInterpolationService',
+  'UserInfoObjectFactory', 'DEFAULT_PROFILE_IMAGE_PATH',
+  function($cookies, $http, $q, $window, UrlInterpolationService,
+      UserInfoObjectFactory, DEFAULT_PROFILE_IMAGE_PATH) {
     var PREFERENCES_DATA_URL = '/preferenceshandler/data';
 
     var userInfo = null;
 
+    // TODO(userIsLoggedIn): Find a way to substitute out constants.DEV_MODE
+    // so that we can check ACSID, SACSID cookies.
+    // if DEV_MODE = false, we have to check ACSID and SACSID cookies.
+    var currentUrl = $window.location.href;
+    var isUserLoggedIn = function() {
+      var userIsLoggedIn_ = $cookies.get('dev_appserver_login') !== undefined ||
+      $cookies.get('ACSID') !== undefined ||
+      $cookies.get('SACSID') !== undefined;
```

Comment:
use `$cookies.hasOwnProperty('…')` instead

Reply:
Okay I will do that